packages/build-utils/src/os.ts

```diff
@@ -0,0 +1,38 @@
+import { readFile } from 'fs-extra';
+import { isErrnoException } from '@vercel/error-utils';
+
+export async function getOsRelease() {
+  try {
+    const data = await readFile('/etc/os-release', 'utf8');
+    return await parseOsRelease(data);
+  } catch (err) {
+    if (isErrnoException(err) && err.code === 'ENOENT') {
+      return null;
+    }
+    throw err;
+  }
+}
+
+export async function parseOsRelease(data: string) {
+  const obj: Record<string, string> = {};
+  // Example file contents:
+  //   NAME="Amazon Linux"
+  //   VERSION="2023"
+  //   ID="amzn"
+  //   ID_LIKE="fedora"
+  for (const line of data.trim().split('\n')) {
+    const m = /(?<key>.*)="(?<value>.*)"/.exec(line);
+    if (!m?.groups) {
+      continue;
+    }
+    obj[m.groups.key] = m.groups.value;
+  }
+  return obj;
+}
+
+export async function getProvidedRuntime() {
+  const os = await getOsRelease();
+  return os?.PRETTY_NAME === 'Amazon Linux 2'
+    ? 'provided.al2'
+    : 'provided.al2023';
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** make fallback explicit

If `getOsRelease()` returns `null`, this logic will return `provided.al2023`, which means that's the fallback value. I think that's correct, but the this function doesn't make it super clear that this happens.

Can we make this explicit?

```suggestion
  const os = await getOsRelease();
  if (!os) {
    return 'provided.al2023';
  }

  return os.PRETTY_NAME === 'Amazon Linux 2'
    ? 'provided.al2'
    : 'provided.al2023';
```